client/my-sites/email/mailboxes/index.tsx

```diff
@@ -129,6 +147,15 @@ const MailboxesManagement = ( {
 		return <MailboxSelectionList domains={ nonWPCOMDomains } />;
 	}
 
+	if ( selectedSiteId && ! hasAtLeastOneMailbox( nonWPCOMDomains ) && newEmail ) {
+		return (
+			<>
+				{ selectedSiteId && <QuerySiteDomains siteId={ selectedSiteId } /> }
+				<ProgressLine statusText={ translate( 'Loading your mailboxes' ) } />
+			</>
+		);
+	}
+
```

Comment:
@fredrikekelund re https://github.com/Automattic/wp-calypso/pull/94766#discussion_r1771139749

This seems to work ok, I moved the notice logic in here too.

Let me know if there is a better way to do this, I'm not quite sure I understand how all this fits together.